app/src/components/utils.js

```diff
@@ -53,6 +56,41 @@ const utils = {
       if (!/^\w+$/.test(k)) throw new Error(`Invalid field name (${k}) in \`context\`.  Only alphanumeric characters and underscore allowed.`);
     });
     return true;
+  },
+
+  validateAttachments: (attachments, attachmentLimit = '5mb') => {
+    if (attachments) {
+      if (!Array.isArray(attachments)) {
+        throw new Error('Invalid value `attachments`');
+      } else {
+        attachments.every(item => {
+          try {
+          //filename
+            if (item.filename === undefined || item.encoding === undefined || item.content === undefined) throw new Error('Attachment is malformed.  Expect filename, encoding, and content fields.');
+
+            if (validator.isEmpty(item.filename)) throw new Error('Attachment `filename` is required');
+            if (validator.isEmpty(item.content)) throw new Error('Attachment `content` is required');
+            //encoding
+            if (!['base64', 'binary', 'hex'].includes(item.encoding)) throw new Error('Invalid value `encoding` for attachment');
+            //content
+            // want to ensure this fits within our expected size limits...
+            // add a little fudge factor here for encoding, otherwise we need to actually write the file out and examine size on disk.
+            const allowedBytes = bytes.parse(attachmentLimit);
+            const acceptableBytes = allowedBytes * 1.05;
```

Comment:
Is there a best practice for how much fudge is acceptable?

Reply:
no idea how much is required.  and it could vary for each encoding.

Reply:
Can run with 5% multiplier for now until we can get a better refinement on the different lengths and encodings.